src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T, K extends keyof T>(
+  items: T[],
+  key: K,
+): GroupsMap<T> {
```

Comment:
will it work that way?
```suggestion
export function groupByKey<T>(
  items: T[],
  key: keyof T,
): GroupsMap<T> {
```